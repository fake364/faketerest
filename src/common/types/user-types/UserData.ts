export type UserData = {
  email: string;
  firstName?: string;
  lastName?: string;
};
