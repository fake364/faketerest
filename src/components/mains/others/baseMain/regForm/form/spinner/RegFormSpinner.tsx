import React from 'react';
import CircleSpinner, {
  SPINNER_COLORS
} from '../../../../../../../common/components/spinner/CircleSpinner';

type Props = {};

const RegFormSpinner: React.FC<Props> = () => {
  return (
    <div className="absolute top-0 left-0 w-full h-full bg-[rgba(0,0,0,0.5)] z-10 flex items-center justify-center">
      <CircleSpinner
        className={'w-[50px] h-[50px]'}
        color={SPINNER_COLORS.WHITE}
      />
    </div>
  );
};

export default RegFormSpinner;
