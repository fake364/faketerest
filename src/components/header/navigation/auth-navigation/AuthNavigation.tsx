import React from 'react';
import LogoTitle from '../common-navigation/logo-title/LogoTitle';
import BaseButton from '../../../../common/components/buttons/base/BaseButton';
import SearchInput from './searchInput/SearchInput';
import CircleIconButton from '../../../../common/components/buttons/CircleIconButton';
import { FaBell } from '@react-icons/all-files/fa/FaBell';
import { FaCommentDots } from '@react-icons/all-files/fa/FaCommentDots';
import ButtonDropdown from '../../../../common/components/buttons/buttonDropdown/ButtonDropdown';
import ButtonDropdownElement from '../../../../common/components/buttons/buttonDropdown/ButtonDropdownElement';
import { useSelector } from 'react-redux';
import { RootState } from '../../../../redux/types';
import { UserData } from '../../../../common/types/user-types/UserData';

type Props = {};

const AuthNavigation: React.FC<Props> = () => {
  const userData: UserData | undefined = useSelector(
    (state: RootState) => state.userData.userData
  );

  return (
    <nav className="sticky py-[18px] px-[12px] flex flex-stretch gap-[14px]">
      <div
        className=" transition-all duration-[300ms] hover:bg-[#f0f0f0] p-[12px] rounded-[50%]
         active:scale-75 items-baseline
      cursor-pointer"
      >
        <LogoTitle titleOff width={24} height={24} />
      </div>
      <BaseButton className="bg-[#000000] text-white rounded-[24px] px-[18px]">
        Главная
      </BaseButton>
      <ButtonDropdown>
        <ButtonDropdownElement onClick={() => {}}>
          Создать пин-идею
        </ButtonDropdownElement>
        <ButtonDropdownElement onClick={() => null}>
          Создать пин
        </ButtonDropdownElement>
      </ButtonDropdown>
      <SearchInput className="flex-1" />
      <CircleIconButton
        Icon={FaBell}
        className="px-[12px]"
        onClick={() => null}
      />
      <CircleIconButton
        Icon={FaCommentDots}
        className="px-[12px]"
        onClick={() => null}
      />
      {userData?.email && (
        <div
          className=" transition-all duration-[300ms] hover:bg-[#f0f0f0] p-[12px] rounded-[50%]
         active:scale-75 items-baseline
      cursor-pointer"
        >
          {userData?.email[0].toUpperCase()}
        </div>
      )}
    </nav>
  );
};

export default AuthNavigation;
