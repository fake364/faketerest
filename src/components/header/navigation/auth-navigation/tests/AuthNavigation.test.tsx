import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import AuthNavigation from '../AuthNavigation';
import RegFormSpinner from '../../../../mains/others/baseMain/regForm/form/spinner/RegFormSpinner';
import { UserData } from '../../../../../common/types/user-types/UserData';

describe('AuthNavigation', () => {
  const mockStore = configureMockStore([]);

  it('should render loading if user data is loading', () => {
    const component = mount(
      <Provider store={mockStore({ userData: { isLoading: true } })}>
        <AuthNavigation />
      </Provider>
    );
    expect(component.find(RegFormSpinner)).toHaveLength(1);
  });

  it('should render loading if user data is null', () => {
    const component = mount(
      <Provider
        store={mockStore({ userData: { isLoading: false, userData: null } })}
      >
        <AuthNavigation />
      </Provider>
    );
    expect(component.find(RegFormSpinner)).toHaveLength(1);
  });

  it('should render auth header with a lot of buttons', () => {
    const component = mount(
      <Provider
        store={mockStore({
          userData: {
            isLoading: false,
            userData: {
              email: 'mail@mail.com',
              firstName: 'First',
              lastName: 'Last',
              username: 'user'
            } as UserData
          }
        })}
      >
        <AuthNavigation />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
