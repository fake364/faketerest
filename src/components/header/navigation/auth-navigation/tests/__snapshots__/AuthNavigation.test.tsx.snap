// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthNavigation should render auth header with a lot of buttons 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <AuthNavigation>
    <nav
      className="sticky py-[18px] px-[12px] flex flex-stretch gap-[14px]"
    >
      <div
        className=" transition-all duration-[300ms] hover:bg-[#f0f0f0] p-[12px] rounded-[50%]
 active:scale-75 items-baseline
 cursor-pointer"
      >
        <LogoTitle
          height={24}
          titleOff={true}
          width={24}
        >
          <div
            className="flex items-center"
          >
            <Image
              alt="Logo"
              height={24}
              src={
                Object {
                  "process": [Function],
                }
              }
              width={24}
            >
              <span
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "boxSizing": "border-box",
                    "display": "inline-block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "overflow": "hidden",
                    "padding": 0,
                    "position": "relative",
                    "width": "initial",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "background": "none",
                      "border": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": "initial",
                      "margin": 0,
                      "maxWidth": "100%",
                      "opacity": 1,
                      "padding": 0,
                      "width": "initial",
                    }
                  }
                >
                  <img
                    alt=""
                    aria-hidden={true}
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                    style={
                      Object {
                        "background": "none",
                        "border": 0,
                        "display": "block",
                        "height": "initial",
                        "margin": 0,
                        "maxWidth": "100%",
                        "opacity": 1,
                        "padding": 0,
                        "width": "initial",
                      }
                    }
                  />
                </span>
                <ImageElement
                  alt="Logo"
                  blurStyle={Object {}}
                  config={
                    Object {
                      "allSizes": Array [
                        16,
                        32,
                        48,
                        64,
                        96,
                        128,
                        256,
                        384,
                        640,
                        750,
                        828,
                        1080,
                        1200,
                        1920,
                        2048,
                        3840,
                      ],
                      "contentSecurityPolicy": "script-src 'none'; frame-src 'none'; sandbox;",
                      "dangerouslyAllowSVG": false,
                      "deviceSizes": Array [
                        640,
                        750,
                        828,
                        1080,
                        1200,
                        1920,
                        2048,
                        3840,
                      ],
                      "disableStaticImages": false,
                      "domains": Array [],
                      "formats": Array [
                        "image/webp",
                      ],
                      "imageSizes": Array [
                        16,
                        32,
                        48,
                        64,
                        96,
                        128,
                        256,
                        384,
                      ],
                      "loader": "default",
                      "minimumCacheTTL": 60,
                      "path": "/_next/image",
                    }
                  }
                  heightInt={24}
                  imgAttributes={
                    Object {
                      "sizes": undefined,
                      "src": "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7",
                      "srcSet": undefined,
                    }
                  }
                  imgStyle={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": undefined,
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                  isLazy={true}
                  isVisible={false}
                  layout="intrinsic"
                  loader={[Function]}
                  onLoadingCompleteRef={
                    Object {
                      "current": undefined,
                    }
                  }
                  placeholder="empty"
                  setBlurComplete={[Function]}
                  setIntersection={[Function]}
                  srcString=""
                  unoptimized={true}
                  widthInt={24}
                >
                  <img
                    alt="Logo"
                    data-nimg="intrinsic"
                    decoding="async"
                    onError={[Function]}
                    onLoad={[Function]}
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style={
                      Object {
                        "border": "none",
                        "bottom": 0,
                        "boxSizing": "border-box",
                        "display": "block",
                        "height": 0,
                        "left": 0,
                        "margin": "auto",
                        "maxHeight": "100%",
                        "maxWidth": "100%",
                        "minHeight": "100%",
                        "minWidth": "100%",
                        "objectFit": undefined,
                        "objectPosition": undefined,
                        "padding": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "width": 0,
                      }
                    }
                  />
                  <noscript>
                    <img
                      alt="Logo"
                      data-nimg="intrinsic"
                      decoding="async"
                      loading="lazy"
                      src=""
                      style={
                        Object {
                          "border": "none",
                          "bottom": 0,
                          "boxSizing": "border-box",
                          "display": "block",
                          "height": 0,
                          "left": 0,
                          "margin": "auto",
                          "maxHeight": "100%",
                          "maxWidth": "100%",
                          "minHeight": "100%",
                          "minWidth": "100%",
                          "objectFit": undefined,
                          "objectPosition": undefined,
                          "padding": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                          "width": 0,
                        }
                      }
                    />
                  </noscript>
                </ImageElement>
              </span>
            </Image>
          </div>
        </LogoTitle>
      </div>
      <BaseButton
        className="bg-[#000000] text-white rounded-[24px] px-[18px] mr-[8px]"
      >
        <button
          className="font-bold active:scale-90 active:brightness-15 bg-[#000000] text-white rounded-[24px] px-[18px] mr-[8px]"
          data-theme="base"
        >
          Главная
        </button>
      </BaseButton>
      <DropdownRootElement
        dropdownClass="left-[-57px]"
        text="Создать"
        variant="text"
      >
        <div
          className="relative flex flex-col self-center"
          onBlur={[Function]}
          tabIndex={0}
        >
          <div
            className="self-center font-medium cursor-pointer"
            onClick={[Function]}
          >
            <span>
              Создать
            </span>
             
            <FaChevronDown
              className="inline-block"
            >
              <IconBase
                attr={
                  Object {
                    "viewBox": "0 0 448 512",
                  }
                }
                className="inline-block"
              >
                <svg
                  className="inline-block"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                    key="0"
                  />
                </svg>
              </IconBase>
            </FaChevronDown>
          </div>
        </div>
      </DropdownRootElement>
      <SearchInput
        className="flex-1"
      >
        <div
          className="relative flex-1"
        >
          <input
            className="bg-[#efefef] rounded-[28px] px-[48px] w-full h-full outline-[4px] focus:outline
 focus:outline-[rgba(0,132,255,0.5)]"
            placeholder="Поиск"
          />
          <div
            className="absolute top-0 h-full flex items-center"
          >
            <FaSearch
              className="ml-[18px] text-[gray]"
            >
              <IconBase
                attr={
                  Object {
                    "viewBox": "0 0 512 512",
                  }
                }
                className="ml-[18px] text-[gray]"
              >
                <svg
                  className="ml-[18px] text-[gray]"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                    key="0"
                  />
                </svg>
              </IconBase>
            </FaSearch>
          </div>
        </div>
      </SearchInput>
      <AuthNavButtons>
        <Tooltip
          text="Уведомления"
        >
          <div
            className="relative [&>#tooltip]:hidden [&>#tooltip]:hover:block flex flex-col items-center justify-center"
          >
            <CircleIconButton
              Icon={[Function]}
              className="px-[12px]"
              onClick={null}
            >
              <div
                className="transition-all duration-[300ms] hover:bg-[#f0f0f0] p-[12px] rounded-[50%] active:scale-75 text-[gray] flex items-center text-[24px] cursor-pointer px-[12px]"
                onClick={null}
              >
                <FaBell>
                  <IconBase
                    attr={
                      Object {
                        "viewBox": "0 0 448 512",
                      }
                    }
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      strokeWidth="0"
                      style={
                        Object {
                          "color": undefined,
                        }
                      }
                      viewBox="0 0 448 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64zm215.39-149.71c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71z"
                        key="0"
                      />
                    </svg>
                  </IconBase>
                </FaBell>
              </div>
            </CircleIconButton>
            <div
              className="absolute bg-black bottom-[-45px] w-fit  p-[6px] text-[#ffffff] rounded-[8px]
 text-[14px]"
              id="tooltip"
            >
              Уведомления
            </div>
          </div>
        </Tooltip>
        <Tooltip
          text="Сообщения"
        >
          <div
            className="relative [&>#tooltip]:hidden [&>#tooltip]:hover:block flex flex-col items-center justify-center"
          >
            <CircleIconButton
              Icon={[Function]}
              className="px-[12px]"
              onClick={null}
            >
              <div
                className="transition-all duration-[300ms] hover:bg-[#f0f0f0] p-[12px] rounded-[50%] active:scale-75 text-[gray] flex items-center text-[24px] cursor-pointer px-[12px]"
                onClick={null}
              >
                <FaCommentDots>
                  <IconBase
                    attr={
                      Object {
                        "viewBox": "0 0 512 512",
                      }
                    }
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      strokeWidth="0"
                      style={
                        Object {
                          "color": undefined,
                        }
                      }
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M256 32C114.6 32 0 125.1 0 240c0 49.6 21.4 95 57 130.7C44.5 421.1 2.7 466 2.2 466.5c-2.2 2.3-2.8 5.7-1.5 8.7S4.8 480 8 480c66.3 0 116-31.8 140.6-51.4 32.7 12.3 69 19.4 107.4 19.4 141.4 0 256-93.1 256-208S397.4 32 256 32zM128 272c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm128 0c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm128 0c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32z"
                        key="0"
                      />
                    </svg>
                  </IconBase>
                </FaCommentDots>
              </div>
            </CircleIconButton>
            <div
              className="absolute bg-black bottom-[-45px] w-fit  p-[6px] text-[#ffffff] rounded-[8px]
 text-[14px]"
              id="tooltip"
            >
              Сообщения
            </div>
          </div>
        </Tooltip>
        <UserRoundButton>
          <Tooltip
            text="Профиль"
          >
            <div
              className="relative [&>#tooltip]:hidden [&>#tooltip]:hover:block flex flex-col items-center justify-center"
            >
              <div
                className=" transition-all duration-[300ms] hover:bg-[#f0f0f0] rounded-[50%]
 active:scale-75 items-baseline px-[10px] h-full flex !items-center justify-center
 cursor-pointer"
              >
                <div
                  className="bg-[#f0f0f0] rounded-[50%] px-[10px] py-[3px]"
                >
                  F
                </div>
              </div>
              <div
                className="absolute bg-black bottom-[-45px] w-fit  p-[6px] text-[#ffffff] rounded-[8px]
 text-[14px]"
                id="tooltip"
              >
                Профиль
              </div>
            </div>
          </Tooltip>
        </UserRoundButton>
      </AuthNavButtons>
      <ParamsArrowDropdown>
        <DropdownRootElement
          Icon={[Function]}
          dropdownClass="right-[0] px-[10px] py-[16px] min-w-[360px]"
          variant="icon"
        >
          <div
            className="relative flex flex-col self-center"
            onBlur={[Function]}
            tabIndex={0}
          >
            <div
              className="self-center font-medium cursor-pointer"
              onClick={[Function]}
            >
              <CircleIconButton
                Icon={[Function]}
                className="px-[12px] text-[14px]"
                onClick={[Function]}
              >
                <div
                  className="transition-all duration-[300ms] hover:bg-[#f0f0f0] p-[12px] rounded-[50%] active:scale-75 text-[gray] flex items-center text-[24px] cursor-pointer px-[12px] text-[14px]"
                  onClick={[Function]}
                >
                  <FaChevronDown>
                    <IconBase
                      attr={
                        Object {
                          "viewBox": "0 0 448 512",
                        }
                      }
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        strokeWidth="0"
                        style={
                          Object {
                            "color": undefined,
                          }
                        }
                        viewBox="0 0 448 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                          key="0"
                        />
                      </svg>
                    </IconBase>
                  </FaChevronDown>
                </div>
              </CircleIconButton>
            </div>
          </div>
        </DropdownRootElement>
      </ParamsArrowDropdown>
    </nav>
  </AuthNavigation>
</Provider>
`;
